app/controllers/admin/api/service_contracts_controller.rb

```diff
@@ -21,14 +29,43 @@ def destroy
     respond_with(service_subscription.unsubscribe(service_contract))
   end
 
+  # Service Subscription Update
+  # PUT /admin/api/accounts/{account_id}/service_contracts/{id}.xml
+  def update
+    service = service_contract.issuer
+    new_plan = service.service_plans.find(service_contract_plan_id)
+    service_contract.change_plan!(new_plan)
+    respond_with(service_contract)
+  end
+
   protected
 
   def account
     @account ||= current_account.buyers.find params.require(:account_id)
   end
 
+  def find_service
+    @service = service
+  end
+
   def service_contract
     @service_contract ||= account.bought_service_contracts.find params.require(:id)
   end
 
+  def service_contract_params
+    params.permit(service_contract: [:plan_id])
+          .fetch(:service_contract).merge(plan: service_plan)
+  end
+
+  def service
+    service ||= accessible_services.find(params[:service_id])
```

Comment:
I mentioned this before, but I'll put it here again - I don't think we need `service_id`, I think it's redundant, because we have a `plan_id`. Another thing is that we'd need to validate this plan ID:
- make sure that it is actually a Service plan
- verify that the user sending the request has permissions on this Service

If these conditions are not met, a proper error must be shown. Currently, if both `service_id` and `plan_id` are set, but there is a mismatch (i.e. the plan doesn't belong to the service specified by the ID), the following error is shown:
```
<?xml version="1.0" encoding="UTF-8"?>
  <errors>
    <error>Plan plan must be a ServicePlan</error>
  </errors>
```

This is confusing, because the plan ID does actually belong to a ServicePlan.

But again, I'd just remove the service ID completely.

Reply:
@mayorova I upadted create action.